osm2streets/src/transform/classify_intersections.rs

```diff
@@ -1,85 +1,229 @@
 use crate::osm::NodeID;
+use crate::types::Movement;
+use crate::Direction;
 use crate::IntersectionComplexity::*;
-use crate::{Direction, IntersectionComplexity, StreetNetwork};
+use crate::{
+    ConflictType, DrivingSide, IntersectionComplexity, RestrictionType, Road, StreetNetwork,
+};
+use std::cmp::{max, min};
+use std::collections::BTreeMap;
 
 /// Determines the initial complexity of all intersections. Intersections marked "Crossing" are
 /// considered "unclassified" and will be updated with a guess, others will be left unchanged.
 pub fn classify_intersections(streets: &mut StreetNetwork) {
-    let mut changes: Vec<(NodeID, IntersectionComplexity)> = Vec::new();
+    let mut changes: Vec<_> = Vec::new();
     for (id, inter) in &streets.intersections {
         if let Crossing = inter.complexity {
             changes.push((*id, guess_complexity(streets, id)));
         }
     }
 
-    for (id, complexity) in changes.into_iter() {
-        streets.intersections.get_mut(&id).unwrap().complexity = complexity;
+    for (id, (complexity, conflict_level, movements)) in changes {
+        let intersection = streets.intersections.get_mut(&id).unwrap();
+        intersection.complexity = complexity;
+        intersection.conflict_level = conflict_level;
+        intersection.movements = movements;
     }
 }
 
 /// Guesses the complexity of the intersection based on the connecting roads and their lanes.
 ///
 /// The existing complexity field is ignored, so be careful how you use the guessed value.
-fn guess_complexity(streets: &StreetNetwork, intersection_id: &NodeID) -> IntersectionComplexity {
-    let roads = streets.roads_per_intersection(*intersection_id);
+pub fn guess_complexity(
+    streets: &StreetNetwork,
+    intersection_id: &NodeID,
+) -> (IntersectionComplexity, ConflictType, Vec<Movement>) {
+    use ConflictType::*;
```

Comment:
nit: Do this once at the top instead

Reply:
Are there ever times when you would find it appropriate to scope these kinds of `use` statements? I take your point that it makes sense at the file scope in this case though.

Reply:
It might be a matter of preference... I think I only do it to avoid unused warnings when doing build flags. Like if you have a web-only `#[cfg(target_arch = "wasm32")]` block and an import that's only used (or available) in there